packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py

```diff
@@ -173,13 +184,22 @@ async def download_path_from_s3(
             raise exceptions.S3InvalidPathError(s3_object)
 
         if file_meta_data.is_directory:
-            assert r_clone_settings  # nosec
-            await r_clone.sync_s3_to_local(
-                r_clone_settings,
-                progress_bar,
-                local_directory_path=local_path,
-                download_s3_link=parse_obj_as(AnyUrl, f"{download_link}"),
-            )
+            if aws_s3_cli_settings:
+                await aws_s3_cli.sync_s3_to_local(
+                    aws_s3_cli_settings,
+                    progress_bar,
+                    local_directory_path=local_path,
+                    download_s3_link=parse_obj_as(AnyUrl, f"{download_link}"),
+                )
+            else:
+                assert r_clone_settings  # nosec
+                await r_clone.sync_s3_to_local(
+                    r_clone_settings,
+                    progress_bar,
+                    local_directory_path=local_path,
+                    download_s3_link=parse_obj_as(AnyUrl, f"{download_link}"),
+                )

```

Comment:
let's go for something more reliable, please use this pattern.
```suggestion
            elif r_clone_settings:
                await r_clone.sync_s3_to_local(
                    r_clone_settings,
                    progress_bar,
                    local_directory_path=local_path,
                    download_s3_link=parse_obj_as(AnyUrl, f"{download_link}"),
                )
            else:
                 msg = "Unexpected configuration"
                 raise RuntimeError(msg)

```